subsys/ipc/ipc_service/backends/Kconfig

```diff
@@ -1,46 +1,37 @@
 # Copyright (c) 2021 Nordic Semiconductor (ASA)
 # SPDX-License-Identifier: Apache-2.0
 
-choice IPC_SERVICE_BACKEND
-	prompt "IPC service backend"
+DT_COMPAT_ZEPHYR_IPC_OPENAMP_STATIC_VRINGS := zephyr,ipc-openamp-static-vrings
+
+config IPC_SERVICE_REG_BACKEND_PRIORITY
+	int "Initialization priority of modules registering IPC backend"
+	default 46
+	help
+	  The backend must be registered before the endpoint register.
 
 config IPC_SERVICE_BACKEND_RPMSG
 	bool "OpenAMP RPMSG backend with static VRINGs"
 	depends on MBOX
+	default $(dt_compat_enabled,$(DT_COMPAT_ZEPHYR_IPC_OPENAMP_STATIC_VRINGS))
```

Comment:
This confuses me. This change is implying that to select this backend we must change the DT. Without this change we are implying that we have to modify the defconfig instead. Now, do we have a strong preference for a way or the other?

Reply:
I don't see any need to enforce the users to explicitly enable this option in configuration file. Afaik ipc-service was ment to be DTS configurable and this change actually does that. I just don't see any advantage of not making it this way. If the user adds `zephyr,ipc-openamp-static-vrings` compatible i believe he/she knows what he/she is doing.
This is more of a proposal rather than hard statement. I know you have objections about that, but i don't understand the justification of why not to do like so.

The only argument against such changes i heard in the past was: Kconfig is for software configuration and DT is for hardware configuration. If so - then configuring backend should never be DT configurable in the first place (imo). But we do accept that, so why not go one step further?

Reply:
> configuring backend should never be DT configurable in the first place (imo)

Right, but also Linux is going to have OpenAMP nodes in DT, so we are not the only ones :)

That said I don't have a strong opinion on this, so this is fine to me.